internal/connect/system.go

```diff
@@ -3,23 +3,56 @@ package connect
 import (
 	"bytes"
 	"fmt"
+	"io"
 	"os"
 	"os/exec"
 	"strings"
 )
 
 var execCommand = exec.Command
 
+// This is a Writer which stores written bytes in a buffer and also echoes them to
+// given output as they come.
+type echoBufferWriter struct {
+	buf *bytes.Buffer
+	out io.Writer
+}
+
+func newEchoBufferWriter(out io.Writer, buf *bytes.Buffer) echoBufferWriter {
+	return echoBufferWriter{buf: buf, out: out}
+}
+
+func (w echoBufferWriter) Write(p []byte) (n int, err error) {
+	w.out.Write(p)
+	return w.buf.Write(p)
+}
+
+var systemEcho bool
+
+// SetSystemEcho toggles piping of executed command's outputs to stdout/stderr
+// returns true if it was enabled before, false otherwise
+func SetSystemEcho(v bool) bool {
+	prev := systemEcho
+	systemEcho = v
+	return prev
+}
+
 func execute(cmd []string, quiet bool, validExitCodes []int) ([]byte, error) {
 	Debug.Printf("Executing: %s Quiet: %v\n", cmd, quiet)
 	var stderr, stdout bytes.Buffer
 	comm := execCommand(cmd[0], cmd[1:]...)
-	comm.Stdout = &stdout
-	comm.Stderr = &stderr
+	if systemEcho {
+		comm.Stdout = newEchoBufferWriter(os.Stdout, &stdout)
```

Comment:
ah nice - I missed that bit in the doc about goroutines reading pipes if these are not os.File.

But I think we could use`io.MultiWriter` and there should be no need for `echoBufferWriter` ?

Reply:
as far as I understand `io.MultiWriter` doesn't interleave the lines but concatenates whole outputs, e.g. first everything from stdout, then everything from stderr.

Reply:
It does what echoBufferWriter does, but in a more general way.

https://github.com/golang/go/blob/50c69cc3a9dfe4f268eebcf3dcbf7f3a93635842/src/io/multi.go#L96-L99
https://github.com/golang/go/blob/50c69cc3a9dfe4f268eebcf3dcbf7f3a93635842/src/io/multi.go#L58-L60

I tried it and it worked
```
	comm.Stdout = io.MultiWriter(os.Stdout, &stdout)
	comm.Stderr = io.MultiWriter(os.Stderr, &stderr)
```

Reply:
Ah, you're right. I was confused by `io.MultiReader` which acts as I described.

Reply:
updated